apps/aecore/src/aec_tx_pool.erl

```diff
@@ -333,12 +333,19 @@ check_nonce_at_hash(Tx, BlockHash) ->
             {error, illegal_nonce};
         true ->
             case aec_chain:get_account_at_hash(Pubkey, BlockHash) of
-                {value, Account} ->
-                    aetx_utils:check_nonce(Account, TxNonce);
                 {error, no_state_trees} ->
```

Comment:
I see two calls of check_nonce_at_hash:
* One when tx is put in mempool, via check_nonce;
* One when building block candidate.

When building block candidate, no_state_trees shall not allow ok i.e. tx in block candidate.

When putting tx in mempool, not sure what behaviour shall be.

Reply:
`no_state_trees` means that the nodeis not started, i.e., is in unit tests. Worst case would be that the transaction would be added while the node is started, and then we get a possibly invalid transaction in the mempool. This transaction will not be added to a block anyway, so it is fine.